### PR TITLE
README for setting up enclave EC2 host

### PR DESCRIPTION
**Pull request by jatanloya**
Readme for #6

### CODE REVIEW

File: doc/setup.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Setting up enclave-enabled EC2 host
+
+1. Go to `Launch an Instance` in the EC2 console
+2. Select `Amazon Linux 2 AMI (HVM) - Kernel 5.10, SSD Volume Type` under AMI
+3. Select `m5a.xlarge` under Instance Type. Complete list of instances which have Nitro Support can be found [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances)
```
Comment (rillian) [inline doc/setup.md:5]: I was going to suggest that `t4g.xlarge` is cheaper, but I just tried it and the console launch interface says it's not supported, [contrary to the documentation](https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave.html#nitro-enclave-reqs). Maybe depends on the region?

Anyway, this seems like a fine recommendation to start people.

File: doc/setup.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Setting up enclave-enabled EC2 host
+
+1. Go to `Launch an Instance` in the EC2 console
+2. Select `Amazon Linux 2 AMI (HVM) - Kernel 5.10, SSD Volume Type` under AMI
```
Comment (rillian) [inline doc/setup.md:4]: I would say rather that most recent Linux distros can work, but installing the tools is easier with Amazon Linux 2023 and Amazon Linux 2.

**Review by rillian — APPROVED** [review]
Thanks for writing this up! One suggested improvement below.

If you want to do more, it would be great to have a section how to set the instance for exploration: installing docker, golang, and aws-nitro-cli packages, starting the services, and so on. Then at the bottom link to `doc/usage.md` for next steps.

File: doc/setup.md (hunk 1 above)
Comment (jatanloya) [reply doc/setup.md:5]: Yes, I have noticed the same issue with other instances, not sure if it is a region issue.

File: doc/setup.md (hunk 2 above)
Comment (jatanloya) [reply doc/setup.md]: I agree this is a better recommendation

File: doc/setup.md [hunk 3]
```diff
@@ -11,3 +11,32 @@
...
+sudo yum -y install docker
+sudo usermod -a -G docker ec2-user
+sudo systemctl enable docker.service
+sudo systemctl start docker.service
+```
+
+2. Install Nitro CLI
+```bash
+sudo amazon-linux-extras install aws-nitro-enclaves-cli -y
+sudo yum install aws-nitro-enclaves-cli-devel -y
```
Comment (rillian) [inline doc/setup.md:29]: On AMZ 2023 this is just `sudo yum install -y aws-nitro-enclaves-cli aws-nitro-enclaves-cli-devel`. Does that work on recent AMZ 2?

**Review by rillian — APPROVED** [review]
Thanks for filling out more of the steps. One last question.

File: doc/setup.md (hunk 3 above)
Comment (jatanloya) [reply doc/setup.md:29]: I tried it, it doesn't work. This is what I had referenced earlier: https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave-cli-install.html

Comment (rillian) [reply doc/setup.md:29]: Ok, thanks for checking.

**Pull request merged**